backend/src/main/java/de/bund/digitalservice/ris/norms/domain/entity/CharacterRange.java

```diff
@@ -0,0 +1,102 @@
+package de.bund.digitalservice.ris.norms.domain.entity;
+
+import de.bund.digitalservice.ris.norms.utils.exceptions.XmlContentException;
+import java.util.regex.Pattern;
+
+/**
+ * Represents a character range within LDML.de.
+ *
+ * @param characterRange the value of the character range as {@link String}
+ */
+public record CharacterRange(String characterRange) {
+  static final int ABSOLUTE_POSITION_OF_START = 0;
+  static final int ABSOLUTE_POSITION_OF_END = 1;
+
+  /**
+   * Returns true if start and end are properly configured
+   *
+   * @return boolean
+   */
+  public boolean isEndGreaterStart() {
+    String[] splitCharacterRange = characterRange().split("-");
+    int start = Integer.parseInt(splitCharacterRange[ABSOLUTE_POSITION_OF_START]);
+    int end = Integer.parseInt(splitCharacterRange[ABSOLUTE_POSITION_OF_END]);
+    return start < end;
+  }
+
+  /**
+   * Returns the start value of the character range
+   *
+   * @return Optional Integer of the start value
+   */
+  public Integer getStart(String articleEid) {
+    isValidCharacterRange(articleEid);
+    String[] splitCharacterRange = characterRange().split("-");
+    return Integer.valueOf(splitCharacterRange[ABSOLUTE_POSITION_OF_START]);
+  }
+
+  /**
+   * Returns the end value of the character range
+   *
+   * @return Optional Integer of the start value
+   */
+  public Integer getEnd(String articleEid) {
+    isValidCharacterRange(articleEid);
+    String[] splitCharacterRange = characterRange().split("-");
+    return Integer.valueOf(splitCharacterRange[ABSOLUTE_POSITION_OF_END]);
+  }
+
+  private void isValidCharacterRange(String articleEId) {
+    final String regex = "^\\d+-\\d+$";
+    final Pattern pattern = Pattern.compile(regex, Pattern.MULTILINE);
+
+    // TODO discuss: I don't see throwing the Exception here
+    if (!pattern.matcher(characterRange()).matches())
+      throw new XmlContentException(
+          "The range (%s) given at article with eId %s is not valid"
+              .formatted(characterRange(), articleEId),
+          null);
+  }
+
+  /** Builder for creating a new {@link CharacterRange}. */
+  public static class Builder {
+    private int start;
+    private int end;
+
+    /**
+     * Sets the start of the character range
+     *
+     * @param start the start of the character range
+     * @return the builder instance
+     */
+    public Builder withStart(int start) {
```

Comment:
everywhere else the methods of builder are just named after the properties directly (so eg. `start` instead of `withStart`)

Reply:
Resolved [here](https://github.com/digitalservicebund/ris-norms/commit/893dd4d048723aed20b9c49bfe2fa5da2740637a)